src/HealthChecks.Network/SftpHealthCheck.cs

```diff
@@ -25,6 +25,11 @@ public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, Canc
                 {
                     var connectionInfo = new ConnectionInfo(item.Host, item.Port, item.UserName, item.AuthenticationMethods.ToArray());
 
+                    if(context.Registration.Timeout > TimeSpan.Zero)

```

Comment:
This healthcheck has no async operation but can use a TimeSpan timeout in the ConnectionInfo class